exchange/exchange.go

```diff
@@ -0,0 +1,35 @@
+package exchange
+
+import "sync"
+
+type queue []func(interface{})
+
+type SendListener interface {
+	Listen(string, func(interface{}))
+	Send(string, interface{})
+}
+
+type Exchange struct {
+	mtx    sync.RWMutex
+	queues map[string]queue
+}
+
+func New() SendListener {
+	return &Exchange{
+		queues: map[string]queue{},
+	}
+}
+
+func (ex *Exchange) Listen(event string, fn func(interface{})) {
+	ex.mtx.Lock()
+	ex.queues[event] = append(ex.queues[event], fn)
+	ex.mtx.Unlock()
+}
+
+func (ex *Exchange) Send(event string, data interface{}) {
+	ex.mtx.RLock()
+	for _, fn := range ex.queues[event] {
+		fn(data)
```

Comment:
Yeah, you're right. The sender shouldn't be blocked and doesn't really care if the listener received it or not. We will just have to make sure to make all the receivers thread safe so that they don't step on themselves.